tests/test_types.py

```diff
@@ -3494,6 +3494,32 @@ class Model(BaseModel):
     }
 
 
+def test_path_like_extra_subtype():
```

Comment:
Test looks great, thanks!

Could you please add some tests that verify validation errors are thrown when invalid data is passed in for each of these parametrized types?